renku/infrastructure/gateway/dataset_gateway.py

```diff
@@ -69,6 +69,7 @@ def remove_tag(self, dataset: Dataset, tag: DatasetTag):
         for t in tags:
             if t.name == tag.name:
                 tags.remove(t)
+                project_context.database._cache.pop(t._p_oid)
```

Comment:
I believe it's cleaner if we add a method named `invalidate_cache` or something similar to the database and call it here. It also make it easier to look for all such uses.

Reply:
It's also safer to call `.pop(t._p_oid, None)` or raise a `RenkuException`

Reply:
actually the db already has a `remove_from_cache` method :see_no_evil: